pkg/dns/treecache.go

```diff
@@ -20,30 +20,21 @@ import (
 	"bytes"
 	"encoding/json"
 	"strings"
-	"sync"
 )
 
-type object interface{}
-
 type TreeCache struct {
 	ChildNodes map[string]*TreeCache
 	Entries    map[string]interface{}
-	rwLock     *sync.RWMutex
 }
 
-func NewTreeCache(m *sync.RWMutex) *TreeCache {
+func NewTreeCache() *TreeCache {
 	return &TreeCache{
 		ChildNodes: make(map[string]*TreeCache),
 		Entries:    make(map[string]interface{}),
-		rwLock:     m,
 	}
 }
 
 func (cache *TreeCache) Serialize() (string, error) {
```

Comment:
I don't think the comment would make sense here. the treecache is not thread-safe.
It could be used by the same thread/go-routine without locking,
or by multiple threads, using a simple mutex or rwmutex. 
it is up to the caller to choose..